msal/src/Microsoft.Identity.Client/MSALTelemetry.cs

```diff
@@ -204,6 +204,21 @@ private List<EventBase> CollateOrphanedEvents(String requestId)
             return orphanedEvents;
         }
 
+        iTelemetry iTelemetry.GetInstance()
+        {
+            return GetInstance();
+        }
+
+        void iTelemetry.StartEvent(string requestId, EventBase eventToStart)
```

Comment:
There is no need to explicitly implement interfaces unless your object implements multiple interfaces and 2 of those interfaces define a method that has the same name. 

https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/interfaces/explicit-interface-implementation

Reply:
This is needed because the itelemetry interface is internal only as we are trying to remove public classes from core. by implementing the interface explicitly,  it allows itelemetry to be internal and used by core while MSALTelemetry can remain public and not be exposed to core.

Reply:
Ah, sneaky... :)

Reply:
Why can't the Implicit implementation give the same (not exposed outside the library)? An interface can be internally implemented

---
In reply to: [208693813](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1173#discussion_r208693813) [](ancestors = 208693813)